### PR TITLE
[BPK-4402] Make bundle cache machine specific

### PR DESCRIPTION
**Pull request by k0nserv**
+ [x] Check this if you have read and followed the [contributing guidelines](https://github.com/Skyscanner/backpack-ios/blob/master/CONTRIBUTING.md)

Remember to include the following changes:
+ [x] `UNRELEASED.md`
+ [x] `README.md`
+ [x] Tests
+ [x] [Screenshotting code](https://github.com/Skyscanner/backpack-ios/blob/master/Example/Backpack%20Screenshot/Screenshots.swift)
+ [x] Adding a component? Remember to expose it in the [main `Backpack.h` header file](https://github.com/Skyscanner/backpack-ios/tree/master/Backpack/Backpack.h)
+ [x] Docs (either update [backpack-docs](https://github.com/Skyscanner/backpack-docs) now, or create a follow up ticket)

_If you are curious about how we review, please read through the [code review guidelines](https://github.com/Skyscanner/backpack/blob/master/CODE_REVIEW_GUIDELINES.md)_

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -72,9 +75,9 @@ jobs:
         uses: actions/cache@v2
         with:
           path: vendor/bundle
-          key: ${{ runner.os}}-gems-${{ hashFiles('**/Gemfile.lock') }}
+          key: ${{ runner.os }}-gems-${{ env.ImageVersion }}-${{ hashFiles('**/Gemfile.lock') }}
```
Comment (georgegillams) [inline .github/workflows/ci.yml:78]: Should we do the same for pod caching too?

Comment (k0nserv) [reply .github/workflows/ci.yml:78]: Mayhaps, but this doesn't actually work so 😓

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -62,19 +65,16 @@ jobs:
       - name: Set Ruby version
         uses: ruby/setup-ruby@v1 # Sets ruby version via `.ruby-version`
 
-      - name: Install bundler
-        run: gem install bundler --version 2.2.3
-
```
Comment (georgegillams) [inline .github/workflows/ci.yml:67]: How do we ensure the correct version of bundler is being used?

Comment (k0nserv) [reply .github/workflows/ci.yml:67]: ruby/setup-ruby handles that by reading the verison from `Gemfile.lock`

image

**Review by georgegillams — APPROVED** [review]

**Pull request merged**